src/valid/mod.rs

```diff
@@ -243,6 +243,121 @@ pub enum ValidationError {
     Corrupted,
 }
 
+#[derive(Clone, Debug, thiserror::Error)]
+#[cfg_attr(test, derive(PartialEq))]
+pub enum ExpressionError {
```

Comment:
Do we have to move this out? Can we make it conditional on the feature flag instead?

Reply:
The problem with making it conditional is that that changes the API of the validator. We want turning off `"validate"` to simply make producing `ModuleInfo` faster, without changing its API.

Reply:
For example, clients that happen to be checking for particular `ExpressionError` variants shouldn't break.

Hmm. Maybe it's *preferable* for any users checking for `FunctionError::Expression` to get a compilation error if the `"validation"` feature is off, since that error will never be generated and users shouldn't be thinking it could happen?

API stability and deliberate API instability both seem to have arguments in their favor. Which do you think is more useful?

Reply:
I think it's natural for feature flags to expose more of the API.

For users that turn on the flag, all enum variants of `ValidationError` will be present, for those who don't they may or may not be there but I don't think those users would exhaustively match the enum anyway.